.github/workflows/ci.yml

```diff
@@ -10,8 +10,8 @@ on:
 
 env:
   # Common versions
-  GO_VERSION: '1.21'
-  GOLANGCI_VERSION: 'v1.56.2'
+  GO_VERSION: '1.23'
+  GOLANGCI_VERSION: 'v1.64.2'
```

Comment:
need to prepare the migration to v2, this can be in a follow-up PR

Reply:
Yeah, there seem to be so many changes that I dreaded updating to v2 in this PR, but needs to get done someday